header/store/store.go

```diff
@@ -221,20 +244,31 @@ func (s *store) Has(_ context.Context, hash tmbytes.HexBytes) (bool, error) {
 	if ok := s.cache.Contains(hash.String()); ok {
 		return ok, nil
 	}
+	// check if the requested header is not yet written on disk
+	if ok := s.pending.Has(hash); ok {
+		return ok, nil
+	}
 
 	return s.ds.Has(datastore.NewKey(hash.String()))
 }
 
-func (s *store) Append(ctx context.Context, headers ...*header.ExtendedHeader) (int, error) {
+func (s *store) Append(ctx context.Context, headers ...*header.ExtendedHeader) (_ int, err error) {
```

Comment:
`err` isn't actually used as an implicit return, unless I'm missing it. This function is also way too big for an implicit return.

```suggestion
func (s *store) Append(ctx context.Context, headers ...*header.ExtendedHeader) (int, error) {
```

Reply:
> This function is also way too big for an implicit return.
Don't get this point tbh.

> err isn't actually used as an implicit return,
Yeah, all returns in the method use explicit return. 

The main reason for the named return param here is to avoid the following:
`var err error`.

Reply:
I agree with John on this.

Reply:
@adlerjohn, I strongly disagree. I am using named var + explicit return which maximally increases readability. There is nothing implicit happening thus nothing for the reader to miss. However, most stakeholders agreed with you, so I have no other choice besides doing this.
> because as-is you're not actually using err as an implicit return and it's not clear to the reader that you didn't miss something